socket_linux.go

```diff
@@ -22,9 +22,9 @@ func createSocketZeroLinger(zeroLinger bool) (fd int, err error) {
 }
 
 // createNonBlockingSocket creates a non-blocking socket with necessary options all set.
-func _createNonBlockingSocket() (int, error) {
```

Comment:
It would be better if we don't use a bool in the parameters to indicate IPv6, since `doSomething(bool)` doesn't give a clue on what the bool means.

Reply:
It's funny -- I took the same approach while just getting it to *work* but I agree passing bools isn't ideal. I'm thinking that the call to `createSocketZeroLinger` can include the actual socket type `AF_INET` or `AF_INET6` to be used.

That way the call stack of socket code can be ignorant of the distinction between ipv6 or not, and the decision of which socket type is appropriate can be made in the right place, the checker code.

Reply:
Sure, a socket type makes much more sense than a bool in this case